PEP 664: Python 3.11 Release Schedule (#2031)
python/peps: pep-0664.rst

Status: Draft
Type: Informational
Content-Type: text/x-rst
Created: 12-Jul-2021
Python-Version: 3.11

Abstract
========

This document describes the development and release schedule for
Python 3.11.  The schedule primarily concerns itself with PEP-sized
items.

.. Small features may be added up to the first beta
   release.  Bugs may be fixed until the final release,
   which is planned for end of October 2021.

Release Manager and Crew
========================

- 3.11 Release Manager: Pablo Galindo Salgado
- Windows installers: Steve Dower
- Mac installers: Ned Deily
- Documentation: Julien Palard

Release Schedule
================

3.11.0 schedule
---------------

Note: the dates below use a 17-month development period that results
in a 12-month release cadence between major versions, as defined by
PEP 602.

Actual:

- 3.11 development begins: Monday, 2021-05-03

Expected:

- 3.11.0 alpha 1: Monday, 2021-10-04
- 3.11.0 alpha 2: Tuesday, 2021-11-02
- 3.11.0 alpha 3: Monday, 2021-12-06
- 3.11.0 alpha 4: Monday, 2022-01-03
- 3.11.0 alpha 5: Wednesday, 2022-02-02
- 3.11.0 alpha 6: Monday, 2022-02-28
- 3.11.0 alpha 7: Tuesday, 2022-04-05

> - 3.11.0 beta 1: Monday, 2022-05-06

Author: ```suggestion
- 3.11.0 beta 1: Friday, 2022-05-06
```

> (No new features beyond this point.)

Reviewer: docutils.utils.SystemMessage: pep-0664.rst:57: (WARNING/2) Bullet list ends without a blank line; unexpected unindent.

Author: 👍

Author: Isn't this already fixed by the last commit? Where is that warning?

PEP: 664
Title: Python 3.11 Release Schedule
Version: $Revision$
Last-Modified: $Date$
Author: Pablo Galindo Salgado <pablogsal@python.org>
Status: Draft
Type: Informational
Content-Type: text/x-rst
Created: 12-Jul-2021
Python-Version: 3.11

Abstract
========

This document describes the development and release schedule for
Python 3.11.  The schedule primarily concerns itself with PEP-sized
items.

.. Small features may be added up to the first beta
   release.  Bugs may be fixed until the final release,
   which is planned for end of October 2021.

Release Manager and Crew
========================

- 3.11 Release Manager: Pablo Galindo Salgado
- Windows installers: Steve Dower
- Mac installers: Ned Deily
- Documentation: Julien Palard

Release Schedule
================

3.11.0 schedule
---------------

Note: the dates below use a 17-month development period that results
in a 12-month release cadence between major versions, as defined by
PEP 602.

Actual:

- 3.11 development begins: Monday, 2021-05-03

Expected:

> - 3.11.0 alpha 1: Monday, 2021-10-04

Reviewer 2: Add a newline here for this to be rendered as a bullet list, at least in the GitHub preview:
```suggestion

- 3.11.0 alpha 1: Monday, 2021-10-04
```

https://github.com/python/peps/blob/1189e291b8985e78eae330dc4d9bbc116b2eb8bf/pep-0664.rst

- 3.11.0 alpha 2: Tuesday, 2021-11-02
- 3.11.0 alpha 3: Monday, 2021-12-06
- 3.11.0 alpha 4: Monday, 2022-01-03
- 3.11.0 alpha 5: Wednesday, 2022-02-02
- 3.11.0 alpha 6: Monday, 2022-02-28
- 3.11.0 alpha 7: Tuesday, 2022-04-05
- 3.11.0 beta 1: Monday, 2022-05-02
(No new features beyond this point.)
- 3.11.0 beta 2: Monday, 2022-05-30
- 3.11.0 beta 3: Thursday, 2022-06-16
- 3.11.0 beta 4: Saturday, 2022-07-09
- 3.11.0 candidate 1: Monday, 2022-08-01
- 3.11.0 candidate 2: Monday, 2022-09-05
- 3.11.0 final:  Monday, 2022-10-03

Subsequent bugfix releases every two months.

3.11 Lifespan
-------------

3.11 will receive bugfix updates approximately every 2 months for
approximately 18 months.  Some time after the release of 3.12.0 final,
the ninth and final 3.11 bugfix update will be released.  After that,
it is expected that security updates (source only) will be released

> until 5 years after the release of 1.11 final, so until approximately

Reviewer 2: ```suggestion
until 5 years after the release of 3.11 final, so until approximately
```

> until 5 years after the release of 1.11 final, so until approximately

Reviewer 2: In fact, should it also be explicit about *which* final? 
```suggestion
until 5 years after the release of 3.11.0 final, so until approximately
```

Status: Draft
Type: Informational
Content-Type: text/x-rst
Created: 12-Jul-2021
Python-Version: 3.11

Abstract
========

This document describes the development and release schedule for
Python 3.11.  The schedule primarily concerns itself with PEP-sized
items.

.. Small features may be added up to the first beta
   release.  Bugs may be fixed until the final release,
   which is planned for end of October 2021.

Release Manager and Crew
========================

- 3.11 Release Manager: Pablo Galindo Salgado
- Windows installers: Steve Dower
- Mac installers: Ned Deily
- Documentation: Julien Palard

Release Schedule
================

3.11.0 schedule
---------------

Note: the dates below use a 17-month development period that results
in a 12-month release cadence between major versions, as defined by
PEP 602.

Actual:

- 3.11 development begins: Monday, 2021-05-03

Expected:

- 3.11.0 alpha 1: Monday, 2021-10-04
- 3.11.0 alpha 2: Tuesday, 2021-11-02
- 3.11.0 alpha 3: Monday, 2021-12-06
- 3.11.0 alpha 4: Monday, 2022-01-03
- 3.11.0 alpha 5: Wednesday, 2022-02-02
- 3.11.0 alpha 6: Monday, 2022-02-28
- 3.11.0 alpha 7: Tuesday, 2022-04-05

> - 3.11.0 beta 1: Monday, 2022-05-02

Reviewer: You might want to bump this a few days as May 2 is the first sprint day of PyCon US 2022:
- https://pycon.blogspot.com/2020/12/announcing-pycon-us-20222023.html

Author: ```suggestion
- 3.11.0 beta 1: Monday, 2022-05-06
```

Reviewer: It's not Monday anymore.

> - 3.11.0 beta 1: Monday, 2022-05-02
> (No new features beyond this point.)
> - 3.11.0 beta 2: Monday, 2022-05-30

Author: ```suggestion
- 3.11.0 beta 1: Monday, 2022-05-02

(No new features beyond this point.)

- 3.11.0 beta 2: Monday, 2022-05-30
```
